pyramid_zipkin/tween.py

```diff
@@ -99,13 +129,14 @@ def zipkin_tween(handler, registry):
     """
     def tween(request):
         zipkin_settings = _get_settings_from_request(request)
-
+        zipkin_port = zipkin_settings.port
         with zipkin_span(
             service_name=zipkin_settings.service_name,
             span_name=zipkin_settings.span_name,
             zipkin_attrs=zipkin_settings.zipkin_attrs,
             transport_handler=zipkin_settings.transport_handler,
-            port=request.server_port,
+            host=zipkin_settings.host,
+            port=(zipkin_port if zipkin_port else request.server_port),
```

Comment:
As we discussed offline, it's probably best to just move the defaulting logic (specifically for `zipkin_port`) into `_get_settings_from_request`